libs/hyperpocket/hyperpocket/cli/tool_export.py

```diff
@@ -0,0 +1,190 @@
+    """Remove hyperpocket import, function_tool decorator, and main execution block."""
+    source_code = re.sub(r"^from hyperpocket\.tool .*?\n", "", source_code, flags=re.MULTILINE)
+    source_code = re.sub(r"@function_tool\([\s\S]*?\)\n", "", source_code, flags=re.MULTILINE)
+    source_code = re.sub(r"\nif __name__ == [\"']__main__[\"']:[\s\S]*", "", source_code, flags=re.MULTILINE)
+    return source_code.strip()
+
+def transform_function_to_pydantic(tool_source_code: str, func: Callable):
+    """Transform function into a Pydantic-based CLI tool."""
+    function_code = extract_function_code(func)
+
+    tree = ast.parse(function_code)
+    function_name = func.func.__name__
+
+    for node in tree.body:
+        if isinstance(node, ast.FunctionDef) and node.name == function_name:
+            params = {arg.arg: arg.annotation.id for arg in node.args.args if isinstance(arg.annotation, ast.Name)}
+            properties = func.argument_json_schema["properties"]
+
+            model_name = function_name.title().replace("_", "") + "Request"
+            model_fields = [
+                f"    {name}: {type_} = Field(..., description=\"{properties[name]['description']}\")"
+                for name, type_ in params.items()
+            ]
+            model_code = f"class {model_name}(BaseModel):\n" + "\n".join(model_fields) + "\n"
+
+            main_code = f"""def main():
+    req = json.load(sys.stdin.buffer)
+    req_typed = {model_name}.model_validate(req)
+    response = {function_name}(**req_typed.model_dump())
+
+    print(json.dumps(response))
+
+
+if __name__ == "__main__":
+    main()
+"""
+            import_code = """import json
+import os
+import sys
+
+from pydantic import BaseModel, Field
+"""
+            return import_code + "\n" + tool_source_code + "\n\n" + model_code + "\n" + main_code
+
+    return None
+
+def extract_function_code(func: Callable) -> str:
+    """Extract function definition as a string."""
+    tool_source_code = inspect.getsource(func.func.__code__)
+    match = re.search(r"def\s+\w+\(.*\):[\s\S]*", tool_source_code)
+    return match.group(0) if match else ""
+
+def generate_tool_template(tool_name: str):
+    """Generate tool template using Click context forwarding."""
+    ctx = click.Context(create_tool_template)
+    ctx.forward(create_tool_template, tool_name=tool_name)
+
+def copy_tool_directory(src_path: Path, tool_name: str):
+    """Copy the tool directory to the new tool directory."""
+    tool_directory_name = tool_name.replace("_", "-")
+    dst_path = src_path.parent / tool_directory_name / tool_name
+
+    if dst_path.exists():
+        shutil.rmtree(dst_path)
+    
+    shutil.copytree(src_path, dst_path)
+    os.remove(dst_path / f"{tool_name}.py")
+    
+    init_file = src_path.parent/ tool_directory_name / tool_name / "__init__.py"    
+    with open(init_file, "w", encoding="utf-8") as f:
+        init_file.write_text(f"from .__main__ import main\n\n__all__ = ['main']")
```

Comment:
```suggestion
    init_file.write_text(f"from .__main__ import main\n\n__all__ = ['main']")
```
write_text opens file, so touching with `open` is not necessary